### PR TITLE
perf(apikey): add apiKey/create action

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/component/driver/apiKey/create.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    const secrets = apiKeyClientSecret.trim().split(",");
+    return secrets.map((secret) => secret.trim());
+  }
+
+  private validateSecret(apiKeySecret: string): boolean {
+    if (typeof apiKeySecret !== "string") {
+      return false;
+    }
+
+    const regExp = /^(\w){10,128}(,\s*\w{10,128})*/g;
```
Comment (SLdragon) [inline packages/fx-core/src/component/driver/apiKey/create.ts:146]: why 128?

Comment (KennethBWSong) [reply packages/fx-core/src/component/driver/apiKey/create.ts:146]: There was a description that client secret should between 10-128 in the design spec, but now it is deleted. Will update this part to max 128.

Comment (KennethBWSong) [reply packages/fx-core/src/component/driver/apiKey/create.ts:146]: I tried a secret length <10 and will get server error and the error message is not clear, so I prefer to keep the validation here.

**Review by SLdragon — DISMISSED** [review]

**Review by jayzhang — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -670,5 +670,15 @@
...
+  "error.core.appIdNotExist": "Cannot find app id: %s. Maybe your current M365 account does not have permission, or the app has alredy been deleted.",
+  "driver.apiKey.description.create": "Create an API key on Developer Portal for authentication in Open API spec.",
+  "driver.aadApp.apiKey.title.create": "Creating API key...",
+  "driver.apiKey.log.startExecuteDriver": "Executing action %s",
+  "driver.apiKey.log.skipCreateApiKey": "Environment variable %s already exists. Skip creating API key.",
+  "driver.apiKey.log.apiKeyNotFound": "Environment variable %s exists but failed to retrieve API key from Developer Portal. Please check if the API key exists manually.",
+  "driver.apiKey.error.nameTooLong": "The name for API key is too long. The maximum length is 128.",
+  "driver.apiKey.error.clientSecretInvalid": "Client secret is invalid. Please make sure: 1. Max %d secrets per API key. 2. Use comma to separate secrets, 3. Length of every secret >= 10 and <= 128",
+  "driver.apiKey.error.domainInvalid": "Domain is invalid. Please make sure provided domain for API key: 1. Max %d domain per API key. 2. Use comma to separate domains",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:681]: Please make sure provided domain for API key: -> Domain for API key should follow:

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -670,5 +670,15 @@
...
-  "error.core.appIdNotExist": "Cannot find app id: %s. Maybe your current M365 account does not have permission, or the app has alredy been deleted."
+  "error.core.appIdNotExist": "Cannot find app id: %s. Maybe your current M365 account does not have permission, or the app has alredy been deleted.",
+  "driver.apiKey.description.create": "Create an API key on Developer Portal for authentication in Open API spec.",
+  "driver.aadApp.apiKey.title.create": "Creating API key...",
+  "driver.apiKey.log.startExecuteDriver": "Executing action %s",
+  "driver.apiKey.log.skipCreateApiKey": "Environment variable %s already exists. Skip creating API key.",
+  "driver.apiKey.log.apiKeyNotFound": "Environment variable %s exists but failed to retrieve API key from Developer Portal. Please check if the API key exists manually.",
+  "driver.apiKey.error.nameTooLong": "The name for API key is too long. The maximum length is 128.",
+  "driver.apiKey.error.clientSecretInvalid": "Client secret is invalid. Please make sure: 1. Max %d secrets per API key. 2. Use comma to separate secrets, 3. Length of every secret >= 10 and <= 128",
```
[the remark this answers is not in the document]
Comment (sffamily) [reply packages/fx-core/resource/package.nls.json:680]: please make sure: -> Client secret should follow:

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -670,5 +670,15 @@
   "error.deploy.AzureStorageSetContainerPropertiesError.Notification": "Unable to set properties of container '%s' in Azure Storage Account '%s' due to error: %s. Refer to the [Output panel](command:fx-extension.showOutputChannel) for more details.",
   "error.deploy.AzureStorageSetContainerPropertiesError": "Unable to set properties of container '%s' in Azure Storage Account '%s' due to error: %s. The error responses from Azure are:\n %s. \nIf the error message specifies the reason, fix the error and try again.",
   "error.core.failedToLoadManifestId": "Unable to load manifest id from path: %s. You must run provision first.",
-  "error.core.appIdNotExist": "Cannot find app id: %s. Maybe your current M365 account does not have permission, or the app has alredy been deleted."
+  "error.core.appIdNotExist": "Cannot find app id: %s. Maybe your current M365 account does not have permission, or the app has alredy been deleted.",
```
[the remark this answers is not in the document]
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json:673]: Updated, thanks!

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -670,5 +670,15 @@
...
+  "driver.apiKey.description.create": "Create an API key on Developer Portal for authentication in Open API spec.",
+  "driver.aadApp.apiKey.title.create": "Creating API key...",
+  "driver.apiKey.log.startExecuteDriver": "Executing action %s",
+  "driver.apiKey.log.skipCreateApiKey": "Environment variable %s already exists. Skip creating API key.",
```
[the remark this answers is not in the document]
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json:677]: Updated, thanks!

File: packages/fx-core/resource/package.nls.json [hunk 6]
```diff
@@ -670,5 +670,15 @@
...
+  "driver.apiKey.log.apiKeyNotFound": "Environment variable %s exists but failed to retrieve API key from Developer Portal. Please check if the API key exists manually.",
```
[the remark this answers is not in the document]
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json:678]: Updated, thanks!

File: packages/fx-core/resource/package.nls.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json:680]: Updated, thanks!

File: packages/fx-core/resource/package.nls.json (hunk 2 above)
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json:681]: Updated, thanks!

**Review by sffamily — APPROVED** [review]

**Review by SLdragon — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by wenytang-ms — APPROVED** [review]

**Review by adashen — APPROVED** [review]

**Pull request merged**